### PR TITLE
Added structured person-name input to relevant mutations.

### PR DESCRIPTION
**Pull request by benwarfield-usds**
Added a NameInput that looks a lot like NameInfo (which looks a lot like PersonName, to be fair). Used it in all the User mutations that take a first/middle/last name, and pushed it through to the service layer with a (reasonably exhaustively tested) translator utility that handles both old and new argument styles.

## Related Issue or Background Info

Sonar keeps complaining (correctly) that we have too many methods on the server side with 20+ string arguments. This is a first step to fixing that.

## Changes Proposed

- Make `name : {firstName: $first, lastName: $last}` a valid substitute for `firstName: $first, lastName: $last` in incoming mutation requests

### CODE REVIEW

File: backend/src/main/java/gov/cdc/usds/simplereport/api/Translators.java [hunk 1]
```diff
@@ -230,4 +232,29 @@ public static String parseState(String s) {
     }
     return symptomsMap;
   }
+
+  public static PersonName consolidateNameArguments(
+      PersonName name, String firstName, String middleName, String lastName, String suffix) {
+    return consolidateNameArguments(name, firstName, middleName, lastName, suffix, false);
+  }
+
+  public static PersonName consolidateNameArguments(
```
Comment (ClydeDroid) [inline backend/src/main/java/gov/cdc/usds/simplereport/api/Translators.java:241]: Is the thought with this that we will eventually convert the frontend to only make calls with `NameInput`, and then remove the translator?

File: backend/src/main/java/gov/cdc/usds/simplereport/api/organization/OrganizationMutationResolver.java [hunk 2]
```diff
@@ -221,14 +228,7 @@ public ApiOrganization createOrganization(
...
             orderingProviderNPI);
-    _aus.createUser(
-        adminEmail,
-        adminFirstName,
-        adminMiddleName,
-        adminLastName,
-        adminSuffix,
-        externalId,
-        Role.ADMIN);
+    _aus.createUser(adminEmail, adminName, externalId, Role.ADMIN);
```
Comment (ClydeDroid) [inline backend/src/main/java/gov/cdc/usds/simplereport/api/organization/OrganizationMutationResolver.java:231]: :ok_hand:

**Review by ClydeDroid — APPROVED** [review]
I dig it! :1st_place_medal:

File: backend/src/main/java/gov/cdc/usds/simplereport/api/organization/OrganizationMutationResolver.java [hunk 3]
```diff
@@ -201,12 +203,17 @@ public ApiOrganization createOrganization(
...
-        new PersonName(
+    providerName = // SPECIAL CASE: MAY BE ALL NULLS/BLANKS
+        Translators.consolidateNameArguments(
+            providerName,
             orderingProviderFirstName,
             orderingProviderMiddleName,
             orderingProviderLastName,
-            orderingProviderSuffix);
+            orderingProviderSuffix,
+            true);
```
Comment (jeskew-gov) [inline backend/src/main/java/gov/cdc/usds/simplereport/api/organization/OrganizationMutationResolver.java:213]: In this case, shouldn't `providerName` just be null?

**Review by jeskew-gov — APPROVED** [review]

File: backend/src/main/java/gov/cdc/usds/simplereport/api/Translators.java (hunk 1 above)
Comment (benwarfield-usds) [reply backend/src/main/java/gov/cdc/usds/simplereport/api/Translators.java:241]: That is my hope, yes! Hence tagging some people who do front-end development so they can decide whether it's worth their time. 😆

File: backend/src/main/java/gov/cdc/usds/simplereport/service/ApiUserService.java [hunk 4]
```diff
@@ -108,20 +88,18 @@ private UserInfo createUserHelper(
...
 
     PersonName nameInfo = apiUser.getNameInfo();
-    nameInfo.setFirstName(firstName);
-    nameInfo.setMiddleName(middleName);
-    nameInfo.setLastName(lastName);
-    nameInfo.setSuffix(suffix);
+    nameInfo.setFirstName(name.getFirstName());
+    nameInfo.setMiddleName(name.getMiddleName());
+    nameInfo.setLastName(name.getLastName());
+    nameInfo.setSuffix(name.getSuffix());
```
[the remark this answers is not in the document]
Comment (benwarfield-usds) [reply backend/src/main/java/gov/cdc/usds/simplereport/service/ApiUserService.java:99]: Arguably, but then some joker could set it to `null` which right now they cannot do. Also I'm not sure how hibernate handles dirty checks on embedded objects, so changing the whole thing could produce weird results (though it shouldn't, I don't think).

**Pull request merged**